packages/demand-bloc-payment/src/providers/adyenErrors.ts

```diff
@@ -45,14 +53,21 @@ export const handleRefusalResponse = (result: HttpResponseOk<PaymentAuthResponse
   if (result.ok && refusalResultCodes.includes(result.body.payload.resultCode as string)) {
     const refusalReasonCode = result.body.payload.refusalReasonCode as string
     const errorMessage = refusalReasons[refusalReasonCode]
-    throw new Error(errorMessage)
+    throw new AdyenProviderError(errorMessage, refusalReasonCode)
   }
   return
 }
 
+export const codes = {
+  AE01: 'AE01',
+  AE02: 'AE02',
+  AE03: 'AE03',
+  AE04: 'AE04',
+}
```

Comment:
I've noticed that we usually prefer plain JS object to TS enums when dealing with error codes. Are there strong reasons for this approach?

Reply:
And what the difference? I think for our purposes it doesn't matter.

Reply:
BTW typescript produces additional JS code when compiles enums, and as a result, bigger bundlesize :nerd_face: